### DISCUSSION TITLE
Using Micronaut as library failing to inject  is failing

### DISCUSSION THREAD

**Discussion by fkeita** [micronaut-projects/micronaut-core#4965] [General]
I created a jpa project called A, I expose my jpa code via services which are singleton.
 I need project B to use project A as library, but injecting Project A services to Project B, causing nullpointer exception. I have no idea what I am doing wrong.  Both projects A and B are micronaut.

annotationProcessor("io.micronaut.data:micronaut-data-processor")
annotationProcessor "io.micronaut:micronaut-inject-java"
    annotationProcessor("io.micronaut.data:micronaut-data-processor")
    implementation("io.micronaut:micronaut-validation")
    implementation("io.micronaut.data:micronaut-data-hibernate-jpa")
	compile "io.micronaut:micronaut-inject"
    compile "io.micronaut:micronaut-runtime"
    compile "io.micronaut:micronaut-inject-java"

**Comment by vs-dsva** [comment]
I am in a similar situation. I want to use micronaut-data part as a library in some code that could work either as a cli app or as a lambda/k8s cron thingie.

**Reply by graemerocher** [reply]
Should be doable no problem most likely it is a build configurable problem, if one of you provide an example of what is not working we can provide feedback on how to fix it
